docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,299 @@
+
+Plus, if the `before`/`beforeEach` setup is unsuccessful, the corresponding test does not run, which saves you time. If `after`/`afterEach` setup yields an error, the following test runs and probably fails due to the unsuccessful setup.
+
+To fulfill your test's state prerequisites, use the `before` and `beforeEach` hooks. Clean up with the `after` and `afterEach` hooks.
+
+Use the `test.before` and `test.after` hooks to set the state that an individual test requires. Use the `fixture.beforeEach` and `fixture.afterEach` to set common state that is required across the board.
+
+## Selectors Strategy
+
+The [TestCafe Studio blog](https://community.devexpress.com/blogs/testcafe/archive/2020/06/10/testcafe-studio-v1-3-0-a-new-way-to-work-with-selectors.aspx#) covers the best ways to work with Selectors.
+
+In general, follow these guidelines when you write the Selectors for your tests.
+
+* Selectors shouldn’t be too specific. Otherwise, you might have to rewrite them after each page modification. For instance, `Selector(‘body’).find(‘div’).nth(5).find(‘p’).nth(3)` must be revised each time the number of elements on the page changes.
+
+* Selectors shouldn't be too generic. Otherwise, they may return different elements after each markup change. For example, `Selector(‘div > button’)` can match multiple elements simultaneously.
+
+* Selectors shouldn’t rely on element parameters that might change. For instance, `Selector('[style*="background-color: red"]')` uses a CSS style that changes often.
+
+* Selectors should remain readable. Selectors should be easy to understand (by you or another developer) for as long as the test is maintained. For instance, it may be difficult to understand which element corresponds to the following selector: `Selector(‘div’).find(‘pre’).nextSibling(-1)`. In contrast, `Selector(‘#topContainer’).find(‘.child’).withText(‘Add item’)` is much easier to read.
+
+* Selectors should reflect the user’s point of view. Since TestCafe supports end-to-end testing, it’s a good idea to build selectors that identify elements as an end-user would. For instance, `Selector(‘form’).find(‘[name=”btn-foo-123”]’)` might be stable, but it is written from the programmer’s perspective rather than from the user’s point of view.
+
+Use custom attributes (like `data-testid`) that are solely devoted to item selection with TestCafe on your page. These attributes are unlikely to change during development and enable you to rewrite your Selectors rarer.
+
+Group the Selectors in a page model. It increases the resilience of your tests and helps remove redundant code.
+
+Use the Selectors extension plugins to work with frameworks. These extensions allow you to create Selectors that are more native to every framework.
```

Comment:
> Use the Selectors extension plugins to work with frameworks.

`to work with frameworks` is very vague. Does it mean to write websites with frameworks?

*Use the Selectors extension plugins for pages built with JavaScript frameworks*?